Products/ZenModel/actions.py

```diff
@@ -400,14 +400,11 @@ def _targetsByTz(self, dmd, targets):
         Take timezone from user property to convert a event time in
         notification and also group targets emails by those timezones.
         """
-        targets_by_timezones = {}
+        tz_targets = {}
         for user in dmd.ZenUsers.getAllUserSettings():
             if user.email in targets:
-                if targets_by_timezones.get(user.timezone):
-                    targets_by_timezones[user.timezone].add(user.email)
-                else:
-                    targets_by_timezones[user.timezone] = set([user.email])
-        return targets_by_timezones
+                tz_targets.setdefault(user.timezone, set([user.email])).add(user.email)
```

Comment:
`tz_targets.setdefault(user.timezone, set()).add(user.email) ` would be better. You don't need to create set with user.email if you are going to try adding it in next instructions anyway.